src/core/execute.c

```diff
@@ -2256,13 +2256,17 @@ static int setup_exec_directory(
                                  * since they all support the private/ symlink logic at least in some
                                  * configurations, see above. */
 
+                                r = chase_symlinks(target, NULL, 0, &resolved, NULL);
```

Comment:
I noticed that maybe this should be `path_simplify()`, otherwise, if e.g., /var/lib is a simlink to somewhere, the issue is still not fixed.

Reply:
Oh, `path_simplify()` does not kill two dots...